File: is_valid_identifier.rb
#!/usr/bin/env ruby
# http://www.codewars.com/kata/is-valid-identifier

def valid_identifier(s)
  digits = (0..9).to_a.map {|e| e.to_s}
  alpha = ("a".."z").to_a
  chars = ["$","_"]
  all = digits + alpha + chars
  if s[0] != "$" && !alpha.include?(s[0].downcase)
    return false
  end
  (1...s.length).each { |c| return false if !all.include?(s[c].downcase) }
  return true
end

["i","wo_rd","b2h"].each { |e| p valid_identifier(e) }
["1i","wo rd","!b2h"].each { |e| p valid_identifier(e) }

File: is_valid_identifier.swift
#!/usr/bin/env swift
// http://www.codewars.com/kata/is-valid-identifier

func validate(s: String) -> Bool {
  guard s.characters.count > 0 else { return false }
  let validCharacters = "abcdefghijklmnopqrstuvwxyz$_".characters.map { String($0) }
  let digits = "0123456789".characters.map { String($0) }
  let stringChar = s.characters.map { String($0) }
  guard validCharacters.contains(stringChar.first!) else { return false }
  for c in stringChar {
    if !validCharacters.contains(c) && !digits.contains(c) {
      return false
    }
  }
  return true
}

func test(list: [String], fn: (String) -> Bool) -> Bool {
  for item in list {
    if !fn(item) {
      return false
    }
  }
  return true
}

let valids = ["i","wo_rd","b2h"]
let invalids = ["1i","wo rd","!b2h"]
print(test(valids, fn: validate))
print(test(invalids, fn: validate))

